### PR TITLE
Add LlamaIndex end to end example

### PR DESCRIPTION
**Pull request by BenWilson2**
**PR Checklist**

- [X] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Add an end-to-end example of using UnityCatalog functions as Agent tools for a LlamaIndex ReAct agent, utilizing MLflow for model management.

### CODE REVIEW

File: ai/integrations/llama_index/unitycatalog-llama_index-with-mlflow.ipynb [hunk 1]
```diff
@@ -0,0 +1,996 @@
...
+    "    model_info = mlflow.llama_index.log_model(\n",
+    "        \"agent.py\",\n",
+    "        artifact_path=\"model\",\n",
+    "        input_example=input_example,\n",
+    "        signature=signature,\n",
+    "        pip_requirements=[\n",
+    "            \"mlflow\",\n",
+    "            \"git+https://github.com/mlflow/mlflow.git\",\n",
+    "            \"git+https://github.com/unitycatalog/unitycatalog.git#subdirectory=ai/core\",\n",
+    "            \"git+https://github.com/unitycatalog/unitycatalog.git#subdirectory=ai/integrations/llama_index\",\n",
```
Comment (serena-ruan) [inline ai/integrations/llama_index/unitycatalog-llama_index-with-mlflow.ipynb:856]: By the time users find this example we should already released! Maybe just replace to normal package name? Same for the %pip install at the top

**Review by serena-ruan — APPROVED** [review]
Overall LGTM!

File: ai/integrations/llama_index/unitycatalog-llama_index-with-mlflow.ipynb [hunk 2]
```diff
@@ -0,0 +1,996 @@
...
+     "nuid": "a78fe7c6-6bdc-4eff-9596-b839e38cdd33",
+     "showTitle": false,
+     "tableResultSettingsMap": {},
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "%pip install -Uqqq git+https://github.com/unitycatalog/unitycatalog.git#subdirectory=ai/core\n",
+    "%pip install -Uqqq git+https://github.com/unitycatalog/unitycatalog.git#subdirectory=ai/integrations/llama_index\n",
```
Comment (BenWilson2) [inline ai/integrations/llama_index/unitycatalog-llama_index-with-mlflow.ipynb:40]:
```suggestion
    "%pip install -Uqqq unitycatalog-ai",
    "%pip install -Uqqq unitycatalog-llamaindex",
```

File: ai/integrations/llama_index/unitycatalog-llama_index-with-mlflow.ipynb [hunk 3]
```diff
@@ -0,0 +1,996 @@
...
+    "# inference should work automatically.\n",
+    "with mlflow.start_run():\n",
+    "    model_info = mlflow.llama_index.log_model(\n",
+    "        \"agent.py\",\n",
+    "        artifact_path=\"model\",\n",
+    "        input_example=input_example,\n",
+    "        signature=signature,\n",
+    "        pip_requirements=[\n",
+    "            \"mlflow\",\n",
+    "            \"git+https://github.com/mlflow/mlflow.git\",\n",
```
Comment (BenWilson2) [inline ai/integrations/llama_index/unitycatalog-llama_index-with-mlflow.ipynb:854]:
```suggestion
```

File: ai/integrations/llama_index/unitycatalog-llama_index-with-mlflow.ipynb (hunk 1 above)
Comment (BenWilson2) [inline ai/integrations/llama_index/unitycatalog-llama_index-with-mlflow.ipynb:856]:
```suggestion
    "            \"unitycatalog-ai\",\n",
    "            \"unitycatalog-llamaindex\",\n",
```

**Pull request merged**